src/validation.cpp

```diff
@@ -3769,18 +3781,9 @@ bool ProcessNewBlock(const CChainParams& chainparams, const std::shared_ptr<cons
         if (fNewBlock) *fNewBlock = false;
         BlockValidationState state;
 
-        // CheckBlock() does not support multi-threaded block validation because CBlock::fChecked can cause data race.
```

Comment:
If you keep CheckBlock as it is can we keep this to inform future refactors ?

Reply:
I think this should either:

- be a comment in `CheckBlock` saying that `cs_main` should usually be held when calling `CheckBlock`
- add an `AssertLockHeld(cs_main)` to `CheckBlock` (and also add that to `FillBlock` and update the callers in bench to hold the lock)
- remove `fChecked` so that `CheckBlock` doesn't need to hold `cs_main`